src/openvpn/options.c

```diff
@@ -3243,7 +3243,7 @@ options_postprocess_mutate_invariant(struct options *options)
         /* delay may only be necessary when we perform DHCP handshake */
         const bool dhcp = (options->tuntap_options.ip_win32_type == IPW32_SET_DHCP_MASQ)
                           || (options->tuntap_options.ip_win32_type == IPW32_SET_ADAPTIVE);
-        if ((options->mode == MODE_POINT_TO_POINT) && dhcp)
+        if ((options->mode == MODE_POINT_TO_POINT) && dhcp && (win32_version_info() <= WIN_VISTA))
```

Comment:
we don't support anything older than Vista anymore and probably not even Vista at this point. So you really would need to remove the whole block instead, instead of basically making is dead code.